cuda_bindings/cuda/bindings/_path_finder/find_nvidia_dynamic_library.py

```diff
@@ -61,88 +59,114 @@ def _find_dll_using_nvidia_bin_dirs(libname, lib_searched_for, error_messages, a
     return None
 
 
-def _get_cuda_paths_info(key, error_messages):
-    env_path_tuple = get_cuda_paths()[key]
-    if not env_path_tuple:
-        error_messages.append(f'Failure obtaining get_cuda_paths()["{key}"]')
-        return None
-    if not env_path_tuple.info:
-        error_messages.append(f'Failure obtaining get_cuda_paths()["{key}"].info')
-        return None
-    return env_path_tuple.info
+def _get_cuda_home():
+    cuda_home = os.environ.get("CUDA_HOME")
+    if cuda_home is None:
+        cuda_home = os.environ.get("CUDA_PATH")
+    return cuda_home
 
 
-def _find_so_using_cudalib_dir(so_basename, error_messages, attachments):
-    cudalib_dir = _get_cuda_paths_info("cudalib_dir", error_messages)
-    if cudalib_dir is None:
+def _find_lib_dir_using_cuda_home(libname):
+    cuda_home = _get_cuda_home()
+    if cuda_home is None:
         return None
-    primary_so_dir = cudalib_dir + "/"
-    candidate_so_dirs = [primary_so_dir]
-    libs = ["/lib/", "/lib64/"]
-    for _ in range(2):
-        alt_dir = libs[0].join(primary_so_dir.rsplit(libs[1], 1))
-        if alt_dir not in candidate_so_dirs:
-            candidate_so_dirs.append(alt_dir)
-        libs.reverse()
-    candidate_so_names = [so_dirname + so_basename for so_dirname in candidate_so_dirs]
-    for so_name in candidate_so_names:
-        if os.path.isfile(so_name):
-            return so_name
-        error_messages.append(f"No such file: {so_name}")
-    for so_dirname in candidate_so_dirs:
-        attachments.append(f'  listdir("{so_dirname}"):')
-        if not os.path.isdir(so_dirname):
-            attachments.append("    DIRECTORY DOES NOT EXIST")
+    if IS_WINDOWS:
+        if libname == "nvvm":  # noqa: SIM108
```

Comment:
Should we just appease the linter instead of noqa'ing these?

Reply:
It seems the if-else blocks are used so that we can have a multi-line list with inline comments

Reply:
Done: commit ab00a872181d0a84cd20bb0459e2fe20b3dbcc37

(It came out looking better than I was anticipating.)